packages/headless/src/api/analytics/insight-analytics.test.ts

```diff
@@ -0,0 +1,120 @@
+        CoveoAnalyticsClient
+    ).toBe(true);
+  });
+
+  it('should be enabled if explicitly specified', () => {
+    const state = buildMockInsightState();
+    state.configuration.analytics.enabled = true;
+
+    expect(
+      configureInsightAnalytics({state, logger}).coveoAnalyticsClient instanceof
+        CoveoAnalyticsClient
+    ).toBe(true);
+  });
+
+  it('should be disabled if explicitly specified', () => {
+    const state = buildMockInsightState();
+    state.configuration.analytics.enabled = false;
+
+    expect(
+      configureInsightAnalytics({state, logger}).coveoAnalyticsClient instanceof
+        CoveoAnalyticsClient
+    ).toBe(false);
+  });
+
+  describe('search analytics provider', () => {
+    const getBaseState = (): StateNeededByInsightAnalyticsProvider => ({
+      configuration: getConfigurationInitialState(),
+    });
+
+    it('should properly return the pipeline from the state', () => {
+      const state = getBaseState();
+      state.pipeline = 'foo';
+      expect(new InsightAnalyticsProvider(state).getPipeline()).toBe('foo');
+    });
+
+    it('should properly return the pipeline from the reponse if not available directly from state', () => {
+      const state = getBaseState();
+      state.pipeline = undefined;
+      state.search = buildMockSearchState({});
+      state.search.response.pipeline = 'foo';
+      expect(new InsightAnalyticsProvider(state).getPipeline()).toBe('foo');
+    });
+
+    it('should properly return facet state', () => {
+      const state = getBaseState();
+      state.facetSet = {the_facet: buildMockFacetRequest({field: 'foo'})};
+      state.search = buildMockSearchState({});
+      state.search.response.facets = [
+        buildMockFacetResponse({
+          field: 'foo',
+          values: [buildMockFacetValue({state: 'selected'})],
+        }),
+      ];
+
+      expect(new InsightAnalyticsProvider(state).getFacetState()[0].field).toBe(
+        'foo'
+      );
+    });
+
+    it('should properly return getSearchEventRequestPayload', () => {
+      const state = getBaseState();
+      state.search = buildMockSearchState({});
+      state.search.response.results = [
+        buildMockResult(),
+        buildMockResult(),
+        buildMockResult(),
+      ];
+      state.query = buildMockQueryState({q: 'foo'});
+      expect(
+        new InsightAnalyticsProvider(state).getSearchEventRequestPayload()
+      ).toMatchObject({
+        queryText: 'foo',
+        responseTime: 0,
+        results: expect.any(Array),
+        numberOfResults: 3,
+      });
+    });
+
+    it('should properly return getSearchUID from searchResponseId if available', () => {
+      const state = getBaseState();
+      state.search = buildMockSearchState({searchResponseId: 'the_id'});
+      state.search.response.searchUid = 'another_id';
+      expect(new InsightAnalyticsProvider(state).getSearchUID()).toEqual(
+        'the_id'
+      );
```

Comment:
Can we have a little bit more descriptive name for this test? It's not easily clear that you're expecting the value from `searchResponseId` even if you modify the `searchUid`.

Reply:
It was handled in the base PR https://github.com/coveo/ui-kit/pull/2147